File: DSA/PriyankaSingh_590016827/day46question2.c
#include <stdio.h>
#define SIZE 100

typedef struct {
    int data[SIZE];
    int front, rear;
} Queue;

void enqueue(Queue *q, int val) {
    q->data[q->rear++] = val;
}

void countEvenOdd(Queue q, int *even, int *odd) {
    *even = 0;
    *odd = 0;
    for (int i = q.front; i < q.rear; i++) {
        if (q.data[i] % 2 == 0)
            (*even)++;
        else
            (*odd)++;
    }
}

int main() {
    Queue q = {.front = 0, .rear = 0};
    int n, val, even, odd;

    printf("Enter number of elements in the queue: ");
    scanf("%d", &n);

    printf("Enter %d integers:\n", n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &val);
        enqueue(&q, val);
    }

    countEvenOdd(q, &even, &odd);

    printf("Even: %d, Odd: %d\n", even, odd);
    return 0;
}

File: DSA/LakshaKundra_590013237/Day_46_Question_2.py
from collections import deque

def count_even_odd(queue):
    even = odd = 0
    size = len(queue)
    for _ in range(size):
        val = queue.popleft()
        if val % 2 == 0:
            even += 1
        else:
            odd += 1
        queue.append(val)
    return even, odd

q1 = deque([1, 2, 3, 4, 5])
e, o = count_even_odd(q1)
print(f"Even: {e}, Odd: {o}")

q2 = deque([6, 8, 10])
e, o = count_even_odd(q2)
print(f"Even: {e}, Odd: {o}")
